src/lib.rs

```diff
@@ -0,0 +1,408 @@
+#![forbid(unsafe_code)]
+
+mod baseline;
+mod check_release;
+mod config;
+mod dump;
+mod manifest;
+mod query;
+mod templating;
+mod util;
+
+use cargo_metadata::PackageId;
+pub use config::*;
+pub use query::*;
+
+use check_release::run_check_release;
+use trustfall_rustdoc::{load_rustdoc, VersionedCrate};
+
+use dump::RustDocCommand;
+use itertools::Itertools;
+use semver::Version;
+use std::collections::HashSet;
+use std::path::{Path, PathBuf};
+
+/// Test a release for semver violations.
+pub struct Check {
+    /// Which packages to analyze.
+    scope: Scope,
+    current: Rustdoc,
+    baseline: Rustdoc,
+    log_level: Option<log::Level>,
+}
+
+pub struct Rustdoc {
+    source: RustdocSource,
+}
+
+impl Rustdoc {
+    /// Use an existing rustdoc file.
+    pub fn from_path(rustdoc_path: impl Into<PathBuf>) -> Self {
+        Self {
+            source: RustdocSource::Rustdoc(rustdoc_path.into()),
+        }
+    }
+
+    /// Generate the rustdoc file from the project root directory, i.e. the directory containing the crate source.
+    /// It can be a workspace or a single package.
+    /// Same as `from_git_revision`, but with the current git revision.
```

Comment:
Is Rust doc comment style to include or omit the parens when naming functions? I genuinely am not sure, we should probably mirror what `std` does here.

Reply:
By looking at [this](https://doc.rust-lang.org/std/env/index.html) page, it seems that the std uses the () convention.
I changed this in a496d420ecd96a1fe338bdc96a92f8960350a7a5